dbt/adapters/firebolt/impl.py

```diff
@@ -28,19 +28,32 @@ class FireboltIndexConfig(dbtClassMixin):
     def render_name(self, relation):
         """
         Name an index according to the following format, joined by `_`:
-        index type, relation name, key/join column, timestamp (unix & UTC)
+        index type, relation name, key/join columns, timestamp (unix & UTC)
         example index name: join_my_model_customer_id_1633504263.
         """
         now_unix = time.mktime(datetime.utcnow().timetuple())
-        spine_col = '_'.join(self.key_column if self.key_column else self.join_column)
+        # If column_names is a list with > 1 members, do a string join,
+        #  otherwise do not. We were getting index names like
+        # join__idx__emf_customers__f_i_r_s_t___n_a_m_e__165093112.
+        if self.key_column:
+            column_names = self.key_column
+        else:
+            # Must be join column
+            column_names = self.join_column
+        if type(column_names) is str:
+            spine_col = column_names
+        elif len(column_names) > 1:
+            spine_col = '_'.join(column_names)
+        else:
+            # column_names is a list with a single member.
+            spine_col = column_names[0]
         inputs = [
-            str(self.index_type),
-            'idx',
+            f'{self.index_type}_idx',
             relation.identifier,
             spine_col,
             str(int(now_unix)),
         ]
-        string = '__'.join(inputs)[:254]
```

Comment:
Why did we have a slice in the first place?

Reply:
It was a legacy of a previous dev and the point wasn't clear to me.

Reply:
I think 256 is the max # of characters you can have for an index name...